### PR TITLE
[#5408] Don't re-verify invalid FITS cards after their values have been replaced

### LINKED ISSUE #5408: FITS headers don't update immediately?
This is an issue noted on https://github.com/astropy/astropy/issues/5396, but is independent of that issue.

In short, given a FITS header with an invalid entry (e.g., a newline within the `'ORIGIN'` value), the following will happen:

```
fh = fits.open(fn)
fh[0].header.tostring() # produces exception
fh[0].header.tostring() # does not produce exception
```

```
>>> h = fits.open('test1n.fits')
>>> h[0].header['origin'] = 'goodval'
>>> w = wcs.WCS(h[0].header)  # <- crashes
# a repeat attempt here would (probably) not crash
```

see: https://github.com/astropy/astropy/issues/5396#issuecomment-252687514, https://github.com/astropy/astropy/issues/5396#issuecomment-252699025, https://github.com/astropy/astropy/issues/5396#issuecomment-252700815, https://github.com/astropy/astropy/issues/5396#issuecomment-252701891

### PR DESCRIPTION
**Pull request by embray**
Fixes #5408.

Fixing this was not so easy because it once again ran up against the huge mess that is FITS card parsing and verification.  If I could I would rewrite the whole thing from scratch.

### CODE REVIEW

File: astropy/io/fits/tests/test_header.py [hunk 1]
```diff
@@ -2388,6 +2388,39 @@ def test_set_keyword_with_space(self):
...
+        such an invalid card.
+        """
+
+        card = fits.Card.fromstring('KW      = INF  / Comment')
+        card.value = 'FIXED'
+        assert tuple(card) == ('KW', 'FIXED', 'Comment')
+        card.verify('fix')
+        assert tuple(card) == ('KW', 'FIXED', 'Comment')
+
+
```
Comment (saimn) [inline astropy/io/fits/tests/test_header.py:2409]: To fix the code style check:
```suggestion

```

Comment (embray) [reply astropy/io/fits/tests/test_header.py:2409]: Oops.  I should really set up my editor to fix that.

**Comment by saimn** [comment]
Looks good, thanks @embray ! Needs a changelog though.

**Comment by embray** [comment]
This was positively reviewed other than missing a changelog entry so I'll go ahead and merge.

**Pull request merged**